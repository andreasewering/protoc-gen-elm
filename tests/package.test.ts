const {
  runPlugin,
  compileElm,
  getGeneratedFileContents,
} = require("./snapshot_test_base");

describe("protoc-gen-elm", () => {
  beforeAll(() => runPlugin("package.proto"));
  const expectedElmFileName = "Any/Package.elm";

  it("generates a valid elm file for package.proto", async () => {
    await compileElm(expectedElmFileName);
  });

  it("generates the expected code for package.proto", async () => {
    const generatedContent = await getGeneratedFileContents(
      expectedElmFileName
    );
    expect(generatedContent).toMatchSnapshot();
  });
});
