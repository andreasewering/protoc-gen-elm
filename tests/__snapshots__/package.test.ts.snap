// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`protoc-gen-elm generates the expected code for package.proto 1`] = `
"{- !!! DO NOT EDIT THIS FILE MANUALLY !!! -}


module Any.Package exposing
    ( Test
    , testDecoder
    , toTestEncoder
    )

{-| ProtoBuf module: \`Any.Package\`

This module was generated automatically using

  - [\`protoc-gen-elm\`](https://www.npmjs.com/package/protoc-gen-elm) 1.0.0
  - \`protoc\` 3.14.0
  - the following specification file: \`package.proto\`

To run it use [\`elm-protocol-buffers\`](https://package.elm-lang.org/packages/eriktim/elm-protocol-buffers/1.1.0) version 1.1.0 or higher.


# Model

@docs Test


# Decoder

@docs testDecoder


# Encoder

@docs toTestEncoder

-}

import Protobuf.Decode as Decode
import Protobuf.Encode as Encode



-- MODEL


{-| \`Test\` message
-}
type alias Test =
    {}



-- DECODER


{-| \`Test\` decoder
-}
testDecoder : Decode.Decoder Test
testDecoder =
    Decode.message Test
        []



-- ENCODER


{-| \`Test\` encoder
-}
toTestEncoder : Test -> Encode.Encoder
toTestEncoder _ =
    Encode.message
        []



-- SETTERS



"
`;
